src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs

```diff
@@ -150,25 +149,23 @@ public static void QueryContextStreamSizes(SafeDeleteContext? securityContext, o
 
         public static void QueryContextConnectionInfo(SafeDeleteSslContext securityContext, ref SslConnectionInfo connectionInfo)
         {
-            connectionInfo.UpdateSslConnectionInfo(securityContext.SslContext);
+            connectionInfo.UpdateSslConnectionInfo((SafeSslHandle)securityContext);
         }
 
-        private static SecurityStatusPal HandshakeInternal(SafeFreeCredentials credential, ref SafeDeleteSslContext? context,
+        private static SecurityStatusPal HandshakeInternal(ref SafeDeleteSslContext? context,
             ReadOnlySpan<byte> inputBuffer, ref byte[]? outputBuffer, SslAuthenticationOptions sslAuthenticationOptions, SelectClientCertificate? clientCertificateSelectionCallback)
         {
-            Debug.Assert(!credential.IsInvalid);
-
             byte[]? output = null;
             int outputSize = 0;
 
             try
             {
                 if ((null == context) || context.IsInvalid)
                 {
```

Comment:
If context.IsInvalid, should we Dispose it?

Reply:
I look through the code and I think the only one way how this could happen after creation is Disposing the context.
I think we added some of the check while back when we had issue with using closed handles.